benchmarks/mmmu/mmmu.py

```diff
@@ -0,0 +1,169 @@
+    mime_type = "image/png"
+    return f"data:{mime_type};base64,{encoded_string}"
+
+
+# Helper function to create content list from record
+def create_content_list(record):
+    content_list = [ContentText(text=record["question"])]
+    for i in range(1, 8):
+        image_field = f"image_{i}"
+        if record.get(image_field):
+            content_list.append(
+                ContentImage(image=image_to_data_url(record[image_field]))
+            )
+    return content_list
+
+
+MULT_CHOICE_PROMPT = r"""
+Answer with the option's letter from the given choices directly.
+{prompt}
+""".strip()
+
+
+def mmmu_task_multiple_choice(dataset):
+    multiple_choice_questions_hf = dataset.filter(
+        lambda example: example["question_type"] == "multiple-choice"
+    )
+    multiple_choice_questions_inspect = MemoryDataset(
+        samples=[
+            record_to_sample_multiple_choice(data)
+            for data in multiple_choice_questions_hf
+        ],
+        name="MMMU-Multiple-Choice",
+        location="MMMU/MMMU",
+    )
+
+    return Task(
+        dataset=multiple_choice_questions_inspect,
+        plan=[prompt_template(MULT_CHOICE_PROMPT), multiple_choice()],
+        scorer=choice(),
+        config=GenerateConfig(temperature=0),
+    )
+
+
+OPEN_PROMPT = r"""
+Answer the question using a single word or phrase.
+{prompt}
+""".strip()
+
+
+def mmmu_task_open(dataset):
+    open_ended_questions_hf = dataset.filter(
+        lambda example: example["question_type"] == "open"
+    )
+    open_ended_questions_inspect = MemoryDataset(
+        samples=[record_to_sample_open_ended(data) for data in open_ended_questions_hf],
+        name="MMMU-Open-Ended",
+        location="MMMU/MMMU",
+    )
+
+    return Task(
+        dataset=open_ended_questions_inspect,
+        plan=[prompt_template(OPEN_PROMPT), generate()],
+        scorer=model_graded_fact(),
+        config=GenerateConfig(temperature=0),
+    )
+
+
+subjects_list = [
+    "Accounting",
+    "Agriculture",
+    "Architecture_and_Engineering",
+    "Art",
+    "Art_Theory",
+    "Basic_Medical_Science",
+    "Biology",
+    "Chemistry",
+    "Clinical_Medicine",
+    "Computer_Science",
+    "Design",
+    "Diagnostics_and_Laboratory_Medicine",
+    "Economics",
+    "Electronics",
+    "Energy_and_Power",
+    "Finance",
+    "Geography",
+    "History",
+    "Literature",
+    "Manage",
+    "Marketing",
+    "Materials",
+    "Math",
+    "Mechanical_Engineering",
+    "Music",
+    "Pharmacy",
+    "Physics",
+    "Psychology",
+    "Public_Health",
+    "Sociology",
+]
+
+
+# Allow multiple-choice/open-ended tasks to be run separately
+@task
+def mmmu_multiple_choice():
+    path = "MMMU/MMMU"
+    datasets = [load_dataset(path, subject, split="dev") for subject in subjects_list]
+    combined_dataset = concatenate_datasets(datasets)
+
+    return mmmu_task_multiple_choice(combined_dataset)
+
+
+# Allow multiple-choice/open-ended tasks to be run separately
+@task
```

Comment:
also, might be an overkill, but maybe it would be nicer to have another two separate files for the two different tasks related code